MDANSE/Src/MDANSE/Framework/Formats/HDFFormat.py

```diff
@@ -181,7 +181,13 @@ def write(
             outputFile = h5py.File.in_memory()
 
         else:
-            filename = Path(filename).with_suffix(extension)
+            user_given_name = Path(filename).name
+            filepath = Path(filename).with_suffix(".mda")
```

Comment:
```suggestion
            filepath = Path(filename).with_suffix(extension)
```

Since this is `HDFFormat.py` and not `MDAFormat.py`, _normally_ it would make sense not to fix the suffix here to ".mda". I am pretty sure that your PR will work just as well if you use the line above, since it is the lines you added below that fix the problem.

Ultimately, the reason why the current version works correctly is because `MDTFormat.write` is never used (!). It seems that all the output filename handling is done by `OutputTrajectoryConfigurator` instead. I will create an issue to deal with it later.